src/pages/[...slug].astro

```diff
@@ -1,38 +1,117 @@
 ---
+
 import { getCollection } from "astro:content";
 
 import Layout from "@/layouts/base.astro";
-import { sys } from "typescript";
 
-export async function getStaticPaths() {
-    const documents = await getCollection("documents");
+export const discordCache = new Map<string,{username:string,pfp:string}|undefined>();
+export const githubCache = new Map<string,{username:string,pfp:string,url:string}|undefined>();
 
-    const documentPaths = documents.map(entry => ({
-    params: {
-      slug: entry.slug === 'index' ? '/' : entry.slug,
+export const getDiscordData = async (userID:string) => {
+  if (discordCache.has(userID)) {
+    return discordCache.get(userID) as {username:string,pfp:string} | undefined;
+  }
```

Comment:
If fetching the avatar fails i.e when a user doesn't exist, the cache entry will remain `undefined`, which will allow for repeat API calls. `null` could be used to special-case this and avoid the situation entirely.